### PR TITLE
ci: find a way to reenable the cherry-picker process

### LINKED ISSUE #3946: Find a way to reenable the cherry-picker process
It's removed in https://github.com/envoyproxy/gateway/pull/3831, because it hardly ever worked.
We need find a way to enable it, so that it could reduce the effort for cherrypicking and releasing minor patch.

### PR DESCRIPTION
**Pull request by Manoramsharma**
**What type of PR is this?**

CI workflow

**What this PR does / why we need it**:

This PR introduces a custom GitHub Actions workflow to automate the cherry-picking of merged pull requests with a specific label into a target branch. This automation simplifies the release management process and ensures that critical changes are propagated to the necessary branches efficiently.

**Which issue(s) this PR fixes:**

Fixes #3946

**Changes made in this PR:**

- Added a new GitHub Actions workflow (`cherry-picker.yml`) to automate the cherry-picking process.
- The workflow is triggered when a pull request targeting the `main` branch is closed.
- The workflow checks if the pull request has the label `cherrypick/release-v1.1` and if it was merged.
- The workflow includes steps to:
  - Checkout the repository.
  - Configure Git with the GitHub Actions bot user.
  - Cherry-pick the merge commit into the target branch `release/v1.1`.
  - Push the changes to the target branch.
- Ensured the workflow adheres to linting standards by removing trailing spaces and properly formatting comments.

**Additional context:**

This workflow helps in reducing manual effort for the release manager by automating the process of cherry-picking merged PRs with specific labels into their respective release branches.

### CODE REVIEW

**Review by zirain — CHANGES_REQUESTED** [review]
directly push to target branch is not allowed

File: .github/workflows/cherry-picker.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Cherry pick
+
+on:
+  pull_request:
+    branches:
+      - main
+    types: ["closed"]
+
+permissions:
+  contents: read
```
Comment (zirain) [inline .github/workflows/cherry-picker.yml:10]: can we create branch with read permission?

Comment (Manoramsharma) [reply .github/workflows/cherry-picker.yml:10]: Thank you for pointing that out. I should have update the permissions in the workflow to ensure that the action can create and push branches. The permissions should be set to `contents: write` and `pull-requests: write`.

Comment (zirain) [reply .github/workflows/cherry-picker.yml:10]: @Manoramsharma thanks for you work, but I think this's not a good issue for begginer(this one need a lot of collaborations with the Envoyproxy org admin).

**Comment by zirain** [comment]
As we discussed, we won't need this for now.
I'd like to close this first, feel free to reopen.

**Pull request closed without merging**